src/averageAges.js

```diff
@@ -20,6 +20,14 @@ function calculateMenAverageAge(people, century) {
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
+
+  const menAges = people.filter(person => century
+    ? century === Math.ceil(person.died / 100) && person.sex === 'm'
+    : person.sex === 'm'
+  ).map(man => man.died - man.born);
```

Comment:
this code looks very messy, too many things are going on here

let's first assign to men variable array with ALL men

and only IF the century is passed we will reassign this variable (I don't recommend using ternary operator here)

then in a separate variable create an array with men's ages